### PR TITLE
format safety keywords on static items

### PR DESCRIPTION
**Pull request by ytmimi**
This includes both `ast::StaticItem` and `ast::StaticForeignItem`. `safety` was added to both `ast::StaticItem` and `ast::SaticForeignItem` in https://github.com/rust-lang/rust/pull/124482.

### CODE REVIEW

File: tests/target/unsafe_extern_blocks.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// See tracking issue for unsafe_extern_blocks
+// https://github.com/rust-lang/rust/issues/123743
+
+#![feature(unsafe_extern_blocks)]
+
+safe static TEST1: i32;
```
Comment (ytmimi) [inline tests/target/unsafe_extern_blocks.rs:6]: I'm not sure that formatting the `safety` here is necessary per this comment https://github.com/rust-lang/rust/pull/124482/files#r1608953719. For now it parses, so I don't think there's an issue, though I'm happy to revert the changes to `StaticParts` if we don't think this is necessary.

Comment (compiler-errors) [reply tests/target/unsafe_extern_blocks.rs:6]: Let's keep it, since it doesn't hurt.

**Review by compiler-errors — APPROVED** [review]

**Review by calebcartwright — APPROVED** [review]

**Comment by calebcartwright** [comment]
probably warrants an addition in the release notes too

**Comment by ytmimi** [comment]
@calebcartwright I'll update the CHANGELOG PR https://github.com/rust-lang/rustfmt/pull/6004 in just a bit to include this

File: src/items.rs [hunk 2]
```diff
@@ -1989,11 +1994,13 @@ fn rewrite_static(
     static_parts: &StaticParts<'_>,
     offset: Indent,
 ) -> Option<String> {
+    println!("rewriting static");
```
Comment (workingjubilee) [inline src/items.rs:1997]: You left in a debug print, it seems.

Comment (RalfJung) [reply src/items.rs:1997]: I am surprised that this did not make any of the tests fail...?

Comment (calebcartwright) [reply src/items.rs:1997]: > I am surprised that this did not make any of the tests fail...?

same (https://github.com/rust-lang/rustfmt/issues/6210#issuecomment-2186797085)

we'll do a postmortem of sorts to review and hopefully identify some preventative measures to prevent a recurrence
